barretenberg/cpp/bootstrap.sh

```diff
@@ -60,41 +65,50 @@ rm -f {build,build-wasm,build-wasm-threads}/CMakeCache.txt
 
 (cd src/barretenberg/world_state_napi && yarn --frozen-lockfile --prefer-offline)
 
-echo "#################################"
-echo "# Building with preset: $PRESET"
-echo "# When running cmake directly, remember to use: --build --preset $PRESET"
-echo "#################################"
-
 export AZTEC_CACHE_REBUILD_PATTERNS=.rebuild_patterns
+HASH=$($ci3/cache/content_hash)
 function build_native {
-  # Build bb with standard preset and world_state_napi with Position Independent code variant
-  cmake --preset $PRESET -DCMAKE_BUILD_TYPE=RelWithAssert
-  cmake --preset $PIC_PRESET -DCMAKE_BUILD_TYPE=RelWithAssert
-  if [ "${CI:-0}" -eq 1 ]; then
+  if $ci3/base/is_build; then
+    echo "#################################"
+    echo "# Building with preset: $PRESET"
+    echo "# When running cmake directly, remember to use: --build --preset $PRESET"
+    echo "#################################"
+    # Build bb with standard preset and world_state_napi with Position Independent code variant
+    cmake --preset $PRESET -DCMAKE_BUILD_TYPE=RelWithAssert
+    cmake --preset $PIC_PRESET -DCMAKE_BUILD_TYPE=RelWithAssert
+    cmake --build --preset $PRESET --target bb
+    cmake --build --preset $PIC_PRESET --target world_state_napi
+    # copy the world_state_napi build artifact over to the world state in yarn-project
+    mkdir -p ../../yarn-project/world-state/build/
+    cp ./build-pic/lib/world_state_napi.node ../../yarn-project/world-state/build/
+
+    $ci3/cache/upload barretenberg-preset-release-$HASH.tar.gz build/bin
+    $ci3/cache/upload barretenberg-preset-release-world-state-$HASH.tar.gz build-pic/lib
+  fi
+  if $ci3/base/is_test && $ci3/cache/should_run barretenberg-test-$HASH; then
+    $ci3/github/endgroup > /dev/tty
```

Comment:
I think we're getting too granular with these markers.
Specifically this won't work right, because we're interleaving all the logs due to parallelising the different builds.
Also, why the /dev/tty needed? voodoo territory.
The markers seem good for being to fairly coarse grain inspect the build. I'm wondering if the entire concept should be limited to the top-level bootstrap around the project loop, which is guaranteed to be serial (at least for now).

Reply:
Just played around with a "denoise" script. Might be something to consider.
![image](https://github.com/user-attachments/assets/eb4cef0d-d18c-4f2b-b269-050e327a8f9a)

Reply:
OK I liked the idea of the markers because once collapsed you can easily see the time taken by every section, open to tweaking